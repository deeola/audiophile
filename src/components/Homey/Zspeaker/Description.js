import React from "react";
import { Desc, Subtitle, Title } from "./Zspeakerstyling";
import {BlackButton } from '../../Buttons/Button';

const Description = ({ title, subtitle }) => {
  return (
    <Desc>
      <Title data-testid='desctitle'>{title}</Title>
      <Subtitle data-testid='descsubtitle'>{subtitle}</Subtitle>
      <BlackButton  label={"SEE PRODUCT"} link='/zx9' />
    </Desc>
  );
};

export default Description;
