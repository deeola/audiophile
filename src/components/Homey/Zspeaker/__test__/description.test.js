import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Description from "../Description";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";

it("renders description box correctly", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Description />
    </BrowserRouter>,
    div
  );
});

it("have a title and subtitle", () => {

    const {getByTestId} = render(<BrowserRouter>
        <Description title={'zspeaker'} subtitle='this is a subtitle' />
      </BrowserRouter>)
      expect(getByTestId('desctitle')).toHaveTextContent('zspeaker')
      expect(getByTestId('descsubtitle')).toHaveTextContent('this is a subtitle')
})

//create  snapshot

it('matches snapshot', () => {
    const tree = renderer.create(
        <BrowserRouter>
        <Description title={'zspeaker'} subtitle='this is a subtitle' />
      </BrowserRouter>
    ).toJSON()

    expect(tree).toMatchSnapshot()
})
