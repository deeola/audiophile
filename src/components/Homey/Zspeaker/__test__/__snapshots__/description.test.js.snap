// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-dkzDqf hrzfoW"
>
  <h1
    className="sc-hKMtZM ctHyQo"
    data-testid="desctitle"
  >
    zspeaker
  </h1>
  <p
    className="sc-eCYdqJ bQVPiA"
    data-testid="descsubtitle"
  >
    this is a subtitle
  </p>
  <a
    className="sc-gKXOVf hXhkdj"
    href="/zx9"
    onClick={[Function]}
  >
    SEE PRODUCT
  </a>
</div>
`;
